src/schema-extension-api.ts

```diff
@@ -92,24 +92,28 @@ class SchemaExtensionAPI implements ExtensionAPI {
 	public requestCustomSchema(resource: string): string[] {
 		const matches = [];
 		for (let customKey of Object.keys(this._customSchemaContributors)) {
-			const contributor = this._customSchemaContributors[customKey];
-			let uri: string;
-			if (contributor.label && workspace.textDocuments) {
-				const labelRegexp = new RegExp(contributor.label, 'g');
-				for (const doc of workspace.textDocuments) {
-					if (doc.uri.toString() === resource) {
-						if (labelRegexp.test(doc.getText())) {
-							uri = contributor.requestSchema(resource);
-							return [uri];
+			try {
+				const contributor = this._customSchemaContributors[customKey];
+				let uri: string;
+				if (contributor.label && workspace.textDocuments) {
+					const labelRegexp = new RegExp(contributor.label, 'g');
+					for (const doc of workspace.textDocuments) {
+						if (doc.uri.toString() === resource) {
+							if (labelRegexp.test(doc.getText())) {
+								uri = contributor.requestSchema(resource);
+								return [uri];
+							}
 						}
 					}
 				}
-			}
 
-			uri = contributor.requestSchema(resource);
+				uri = contributor.requestSchema(resource);
 
-			if (uri) {
-				matches.push(uri);
+				if (uri) {
+					matches.push(uri);
+				}
+			} catch (error) {
+				console.log(`Error thrown while requesting schema ` + error);
```

Comment:
is there an output channel where this error can be logged? or we keep it at console level so only developers will see it?

Reply:
+1 to have output channel, we need provide, as much as possible, information when something goes wrong.

Reply:
I think it is better to handle it in a separate PR as it is a distinct feature to have an outputchannel. i created https://github.com/redhat-developer/vscode-yaml/pull/327